### ISSUE TITLE
Configing your own reverse proxy: docker containers bond to 127.0.0.1 only

### ISSUE THREAD

**Issue by CJerkins** [spantaleev/matrix-docker-ansible-deploy#171]
I think there could be a vars created to provide an option to bind the container to loopback or any ip or a specific ip. Use case would be if I were to separate my web server from synapse in a separate VM the bonded container would create a connection issue.

**Comment by spantaleev** [comment]
We could introduce additional variables to solve this.

Example (with their default values):

```yaml
matrix_synapse_container_client_api_host_bind_addr: "127.0.0.1:8008"

matrix_synapse_container_federation_api_plain_host_bind_addr: "127.0.0.1:8048"

# Untested if Docker's `-p` flag likes `0.0.0.0` or if we should just use `"8448"` below
matrix_synapse_container_federation_api_secure_host_bind_addr: "0.0.0.0:8448"

matrix_synapse_container_metrics_host_bind_addr: "127.0.0.1:{{ matrix_synapse_metrics_port }}"
```

And then we adjust the `.service` file to use these variables instead of hardcoding values like it does now.

We should also do the same for all other services (not just Synapse).

If you've got time and wish to work on this yourself, please go ahead!

**Comment by spantaleev** [comment]
I've been hoping to get rid of the `matrix_*_expose_port` variables in favor of something more flexible for a long time. Thank you for prompting me to do finally do it! :+1: 

I've just implemented this in ab59cc5.
